src/documentation/documentationengine/documentationviewer.cpp

```diff
@@ -0,0 +1,227 @@
+#include "documentationviewer.h"
+
+#include <QDir>
+#include <QMenu>
+#include <QToolTip>
+#include <QToolBar>
+#include <QMenuBar>
+#include <QLineEdit>
+#include <QStatusBar>
+#include <QProgressBar>
+#include <QWebEngineView>
+#include <QWebEnginePage>
+#include <QStackedWidget>
+#include <QStandardPaths>
+#include <QWebEngineHistory>
+
+#include "connectionmanager.h"
+#include "documentationengine.h"
+#include "documentationsearch.h"
+#include "htmlcontentgenerator.h"
+
+DocumentationViewer::DocumentationViewer(QWidget *parent)
+    : QMainWindow (parent)
+{
+    mDocumentationEngine = new DocumentationEngine(this);
+    mConnectionManager = new ConnectionManager(this);
+
+    mUrlEdit = new QLineEdit(this);
+    mProgressBar = new QProgressBar(this);
+    mWebView = new QWebEngineView(this);
+    mWebView->load(QUrl("http://www.google.com"));
+    setCentralWidget(mWebView);
+
+    mBottomToolBar = new QToolBar(this);
+    addToolBar(Qt::BottomToolBarArea, mBottomToolBar);
+
+    mStackedWidget = new QStackedWidget(this);
+
+    setWindowTitle("C++ Documentation");
+    resize(800, 600);
+
```

Comment:
I would rather bind size to screen size

// receive current screen geometry
    QScreen *screen = qApp->screens().at(0);
    QRect screenGeometry = screen->geometry();

    // resizing & centring dialog
    resize(screenGeometry.width() / 3,
           screenGeometry.height() / 3);